### PR TITLE
HPCC-21150 Report Spark-thor in ECLWatch System Server page

### PR DESCRIPTION
**Pull request by wangkx**
1. Add code to retrieve Spark-thor configuration from environment
settings;
2. Add code to display Spark-thor in ECLWatch System Server page,
including a web link to view spark cluster information, a web link
to thor, and a web lonk to Spark-thor log.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by wangkx** [comment]
@rpastrana please review.

File: common/environment/environment.cpp [hunk 1]
```diff
@@ -1103,6 +1159,57 @@ class CConstDropZoneInfo : public CConstEnvBase, implements IConstDropZoneInfo
...
+class CConstSparkThorInfo : public CConstEnvBase, implements IConstSparkThorInfo
+{
+public:
+    IMPLEMENT_IINTERFACE;
+    IMPLEMENT_ICONSTENVBASE;
+    CConstSparkThorInfo(CLocalEnvironment *env, IPropertyTree *root) : CConstEnvBase(env, root) {}
+
+    virtual IStringVal &getBuild(IStringVal &str) const
+    {
+        str.set(root->queryProp("@build"));
```
Comment (rpastrana) [inline common/environment/environment.cpp]: is it possible root doesn't contain any of these attributes? if so is it worthwhile to verify if hasprop before setting the str?

File: esp/eclwatch/ws_XSLT/nls/en/hpcc.xml [hunk 2]
```diff
@@ -156,6 +157,7 @@
 <st id='ViewingPage'>Viewing page</st>
 <st id='ViewLogFile'>View Log File</st>
 <st id='ViewAuditLogFile'>View Audit Log File</st>
+<st id='ViewSparkClusterInfo'>View Spark Cluster Information</st>
```
Comment (rpastrana) [inline esp/eclwatch/ws_XSLT/nls/en/hpcc.xml]: Depending on the context in which this message is used, it might need to to change to:
"View Spark-Thor Cluster Information"
or
"View Integrated Spark Cluster Information"

File: esp/eclwatch/ws_XSLT/services.xslt [hunk 3]
```diff
@@ -234,6 +234,16 @@
                 var url = protocol + "://" + window.location.hostname + ":" + port;
                 parent.document.location.href = url;
             }
+            function showPopup(protocol, ip, port)
+            {
+                var url = protocol + "://" + ip + ":" + port;
```
Comment (rpastrana) [inline esp/eclwatch/ws_XSLT/services.xslt]: can we default to http in case protocol is empty?

File: esp/scm/ws_topology.ecm [hunk 4]
```diff
@@ -31,7 +32,10 @@ ESPStruct TpMachine
     string Path;
     int    Port;
     [min_ver("1.18")] int    ProcessNumber;
+    [min_ver("1.28")] unsigned NicSpeedMbitSec;
```
Comment (rpastrana) [inline esp/scm/ws_topology.ecm]: minor, doesn't seem related

**Review by rpastrana — COMMENTED** [review]
@wangkx looks good, just a couple of minor comments to consider.

File: common/environment/environment.cpp (hunk 1 above)
Comment (wangkx) [reply common/environment/environment.cpp]: The set() can handle a nullptr. I do not need to verify here.

File: esp/eclwatch/ws_XSLT/nls/en/hpcc.xml (hunk 2 above)
Comment (wangkx) [reply esp/eclwatch/ws_XSLT/nls/en/hpcc.xml]: Ok, I will change it to "View Spark-Thor Cluster Information".

File: esp/eclwatch/ws_XSLT/services.xslt (hunk 3 above)
Comment (wangkx) [reply esp/eclwatch/ws_XSLT/services.xslt]: Will do.

File: esp/scm/ws_topology.ecm (hunk 4 above)
Comment (wangkx) [reply esp/scm/ws_topology.ecm]: Ok, I am going to move another PR.

**Review by rpastrana — APPROVED** [review]
@wangkx looks good

**Comment by wangkx** [comment]
@rpastrana please review my changes.

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.
